portal/common/lib/bcs_data_parsing.ts

```diff
@@ -2,20 +2,27 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { bcs, BcsType } from "@mysten/bcs";
-import { fromHEX, toHEX } from "@mysten/sui/utils";
+import { fromHEX, toHEX, toB64 } from "@mysten/sui/utils";
 import { base64UrlSafeEncode } from "./url_safe_base64";
 
 const Address = bcs.bytes(32).transform({
     input: (id: string) => fromHEX(id),
     output: (id) => toHEX(id),
 });
 
-// Blob IDs are represented on chain as u256, but serialized in URLs as URL-safe Base64.
+// Blob IDs & hashes are represented on chain as u256, but serialized in URLs as URL-safe Base64.
 const BLOB_ID = bcs.u256().transform({
     input: (id: string) => id,
     output: (id) => base64UrlSafeEncode(bcs.u256().serialize(id).toBytes()),
 });
 
+// Different than BLOB_ID, because we don't want this to be URL-safe;
+// otherwise, it will mess up with the checksum results.
+const U256 = bcs.u256().transform({
```

Comment:
Nit: something like this?
```suggestion
const DATA_HASH = bcs.u256().transform({
```